### PR TITLE
Improve Serializer Factory

### PR DESCRIPTION
**Pull request by TeslaCN**
Ref #1284 

Changes proposed in this pull request:
- Fail fast if (de)serializer not found.

### CODE REVIEW

File: elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/deserializer/RequestBodyDeserializerFactory.java [hunk 1]
```diff
@@ -84,7 +84,10 @@ public static RequestBodyDeserializer getRequestBodyDeserializer(final String co
                 deserializer = REQUEST_BODY_DESERIALIZERS.get(contentType);
             }
         }
-        return deserializer != MISSING_DESERIALIZER ? deserializer : null;
+        if (MISSING_DESERIALIZER == deserializer) {
+            throw new RequestBodyDeserializerNotFoundException(contentType);
+        }
+        return deserializer;
```
Comment (terrymanu) [inline elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/deserializer/RequestBodyDeserializerFactory.java]: The result variable should named as `result`

File: elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/deserializer/RequestBodyDeserializerNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package org.apache.shardingsphere.elasticjob.restful.deserializer;
+
+import java.text.MessageFormat;
+
+/**
+ * {@link RequestBodyDeserializer} not found for specific MIME type.
+ */
+public class RequestBodyDeserializerNotFoundException extends RuntimeException {
```
Comment (terrymanu) [inline elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/deserializer/RequestBodyDeserializerNotFoundException.java]: Please add final if class is not design for extension

File: elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/serializer/ResponseBodySerializerFactory.java [hunk 3]
```diff
@@ -84,7 +84,10 @@ public static ResponseBodySerializer getResponseBodySerializer(final String cont
                 serializer = RESPONSE_BODY_SERIALIZERS.get(contentType);
             }
         }
-        return serializer != MISSING_SERIALIZER ? serializer : null;
+        if (MISSING_SERIALIZER == serializer) {
+            throw new ResponseBodySerializerNotFoundException(contentType);
+        }
+        return serializer;
```
Comment (terrymanu) [inline elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/serializer/ResponseBodySerializerFactory.java]: The result variable should named as `result`

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/serializer/ResponseBodySerializerNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package org.apache.shardingsphere.elasticjob.restful.serializer;
+
+import java.text.MessageFormat;
+
+/**
+ * {@link ResponseBodySerializer} not found for specific MIME type.
+ */
+public class ResponseBodySerializerNotFoundException extends RuntimeException {
```
Comment (Technoboy-) [inline elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/serializer/ResponseBodySerializerNotFoundException.java]: Add final

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**